File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack=[]
    for i in range(len(s)):
        if not stack:
            if s[i]=='(':
                stack.append(s[i])
            else:
                return False
        else:
            if s[i]==')':
                if stack[-1]!='(':
                    return False
                else:
                    stack.pop()
            else:
                stack.append(s[i])
    if not stack:
        return True
    return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {

        Stack<Character> stck = new Stack<>();

        for (int i = 0; i < s.length(); i++) {

            if (s.charAt(i) == '(') {
                stck.push(s.charAt(i));
            } else {

                if (stck.isEmpty()) {
                    return false;
                }

                stck.pop();
            }
        }

        return stck.isEmpty();
    }
}